src/main/java/org/joshsim/lang/interpret/fragment/discover/ConfigValueFragment.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Fragment containing a single configuration value for discovery purposes.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.lang.interpret.fragment.discover;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import org.joshsim.engine.value.type.EngineValue;
+import org.joshsim.lang.interpret.fragment.FragmentType;
+
+/**
+ * Fragment implementation that holds information about a single config value.
+ *
+ * <p>This fragment type is used during configuration discovery to hold parsed
+ * information about a single config value expression, such as a default value
+ * that includes both a number and units.</p>
+ */
+public class ConfigValueFragment extends ConfigDiscoverabilityFragment {
+  private final Optional<String> name;
+  private final Optional<BigDecimal> number;
+  private final Optional<String> units;
+  private final Optional<EngineValue> engineValue;
+
+  /**
+   * Creates a new fragment for a simple name without additional value information.
+   *
+   * @param name The name string
+   * @throws IllegalArgumentException if name is null or empty
+   */
+  public ConfigValueFragment(String name) {
+    if (name == null || name.trim().isEmpty()) {
```

Comment:
I think it's ok to drop this if statement and exception. It is inconsistent with the rest of the codebase.